gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,355 @@
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager project metadata object.
+ * 
+ * A Project is a high-level Google Cloud Platform entity. It is a container for ACLs, APIs,
+ * AppEngine Apps, VMs, and other Google Cloud Platform resources.
+ */
+public class ProjectInfo implements Serializable {
+
+  private static final long serialVersionUID = 9148970963697734236L;
+  private final String name;
+  private final String projectId;
+  private final Map<String, String> labels;
+  private final Long projectNumber;
+  private final State state;
+  private final Long createTimeMillis;
+  private final ResourceId parent;
+
+  /**
+   * The project lifecycle states.
+   */
+  public enum State {
+    /**
+     * Only used/useful for distinguishing unset values
+     */
+    LIFECYCLE_STATE_UNSPECIFIED,
+
+    /**
+     * The normal and active state
+     */
+    ACTIVE,
+
+    /**
+     * The project has been marked for deletion by the user or by the system (Google Cloud
+     * Platform). This can generally be reversed by calling {@link ResourceManager#undelete}.
+     */
+    DELETE_REQUESTED,
+
+    /**
+     * the process of deleting the project has begun. Reversing the deletion is no longer possible.
+     */
+    DELETE_IN_PROGRESS
+  }
+
+  static class ResourceId implements Serializable {
+
+    private static final long serialVersionUID = -325199985993344726L;
+
+    private final String id;
+    private final String type;
+
+    ResourceId(String id, String type) {
+      this.id = checkNotNull(id);
+      this.type = checkNotNull(type);
+    }
+
+    String id() {
+      return id;
+    }
+
+    String type() {
+      return type;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return obj instanceof ResourceId && Objects.equals(toPb(), ((ResourceId) obj).toPb());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(id, type);
+    }
+
+    com.google.api.services.cloudresourcemanager.model.ResourceId toPb() {
+      com.google.api.services.cloudresourcemanager.model.ResourceId resourceIdPb =
+          new com.google.api.services.cloudresourcemanager.model.ResourceId();
+      resourceIdPb.setId(id);
+      resourceIdPb.setType(type.toString().toLowerCase());
+      return resourceIdPb;
+    }
+
+    static ResourceId fromPb(
+        com.google.api.services.cloudresourcemanager.model.ResourceId resourceIdPb) {
+      return new ResourceId(resourceIdPb.getId(), resourceIdPb.getType());
+    }
+  }
+
+  public static class Builder {
+
+    private String name;
+    private String projectId;
+    private Map<String, String> labels = new HashMap<>();
+    private Long projectNumber;
+    private State state;
+    private Long createTimeMillis;
+    private ResourceId parent;
+
+    private Builder() {
+    }
+
+    Builder(ProjectInfo info) {
+      this.name = info.name;
+      this.projectId = info.projectId;
+      this.labels = Maps.newHashMap(checkNotNull(info.labels));
```

Comment:
Nit, `labels.putAll(info.lables)` 
- no need for the checkNotNull (as you ProjectInfo is guaranteed not to have a null labels).
- You can keep the Maps.newHashMap, but then you want to move the `new HashMap<>()` to the empty constructor (my suggestion was before we added the `Builder(ProjectInfo)`).